[libwebp] 1.6.0 build failure on wasm32-emscripten

### Operating system

iMAC intel core

### Compiler

wasm32

### Steps to reproduce the behavior

```Shell
vcpkg install libwebp:wasm32-emscripten
```

### Failure logs

huangbeiao@BEIAOHUANG-MC0 UGit % vcpkg install libwebp:wasm32-emscripten                                                                              
Computing installation plan...
The following packages will be built and installed:
    libwebp[core,libwebpmux,nearlossless,simd]:wasm32-emscripten@1.6.0
Detecting compiler hash for triplet wasm32-emscripten...
Compiler found: /Users/huangbeiao/Documents/emsdk/upstream/emscripten/em++
Restored 0 package(s) from /Users/huangbeiao/.cache/vcpkg/archives in 13.9 us. Use --debug to see more details.
Installing 1/1 libwebp[core,libwebpmux,nearlossless,simd]:wasm32-emscripten@1.6.0...
Building libwebp[core,libwebpmux,nearlossless,simd]:wasm32-emscripten@1.6.0...
/Users/huangbeiao/Documents/UGit/vcpkg/triplets/community/wasm32-emscripten.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
-- Using cached webmproject-libwebp-v1.6.0.tar.gz
-- Cleaning sources at /Users/huangbeiao/Documents/UGit/vcpkg/buildtrees/libwebp/src/v1.6.0-7e52d1be79.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/huangbeiao/Documents/UGit/vcpkg/downloads/webmproject-libwebp-v1.6.0.tar.gz
-- Applying patch 0002-cmake-config.patch
-- Applying patch 0003-simd.patch
-- Applying patch 0008-sdl.patch
-- Using source at /Users/huangbeiao/Documents/UGit/vcpkg/buildtrees/libwebp/src/v1.6.0-7e52d1be79.clean
-- Found external ninja('1.12.1').
-- Configuring wasm32-emscripten
-- Building wasm32-emscripten-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/local/bin/cmake --build . --config Debug --target install -- -v -j17
    Working Directory: /Users/huangbeiao/Documents/UGit/vcpkg/buildtrees/libwebp/wasm32-emscripten-dbg
    See logs for more information:
      /Users/huangbeiao/Documents/UGit/vcpkg/buildtrees/libwebp/install-wasm32-emscripten-dbg-out.log

Call Stack (most recent call first):
  installed/x64-osx/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-osx/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/libwebp/portfile.cmake:53 (vcpkg_cmake_install)
  scripts/ports.cmake:206 (include)


error: building libwebp:wasm32-emscripten failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle libwebp:wasm32-emscripten: 20 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libwebp
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[libwebp]+Build+error+on+wasm32-emscripten&body=Copy%20issue%20body%20from%20%2FUsers%2Fhuangbeiao%2FDocuments%2FUGit%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

[install-wasm32-emscripten-dbg-out.log](https://github.com/user-attachments/files/21936599/install-wasm32-emscripten-dbg-out.log)

### Additional context

_No response_

> "sse41" in the filename, is this really meant to be compiled for wasm?
> 
> ```
> FAILED: CMakeFiles/webpdsp.dir/src/dsp/enc_sse41.c.o 
> /Users/huangbeiao/Documents/emsdk/upstream/emscripten/emcc -DHAVE_CONFIG_H -I/Users/huangbeiao/Documents/UGit/vcpkg/buildtrees/libwebp/wasm32-emscripten-dbg -I/Users/huangbeiao/Documents/UGit/vcpkg/buildtrees/libwebp/src/v1.6.0-7e52d1be79.clean -pthread -g -fPIC -fvisibility=hidden -Wall -msimd128 -msse4.1 -MD -MT CMakeFiles/webpdsp.dir/src/dsp/enc_sse41.c.o -MF CMakeFiles/webpdsp.dir/src/dsp/enc_sse41.c.o.d -o CMakeFiles/webpdsp.dir/src/dsp/enc_sse41.c.o -c /Users/huangbeiao/Documents/UGit/vcpkg/buildtrees/libwebp/src/v1.6.0-7e52d1be79.clean/src/dsp/enc_sse41.c
> /Users/huangbeiao/Documents/UGit/vcpkg/buildtrees/libwebp/src/v1.6.0-7e52d1be79.clean/src/dsp/enc_sse41.c:131:5: error: call to undeclared function 'VP8Transpose_2_4x4_16b'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
>   131 |     VP8Transpose_2_4x4_16b(&b0, &b1, &b2, &b3, &tmp_0, &tmp_1, &tmp_2, &tmp_3);
>       |     ^
> 1 error generated.
> ```

Yes, I just re-ran the command _**vcpkg install libwebp:wasm32-emscripten**_ using the newly cloned vcpkg repository, and I still encounter the same error. I don’t have this issue when compiling other libraries (e.g., harfbuzz, libjpeg-turbo, gtest, etc.), so my environment should be fine. I wanted to ask—do you not get an error when running the same command (_vcpkg install libwebp:wasm32-emscripten_)?

--- a/ports/libwebp/vcpkg.json
+++ b/ports/libwebp/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "libwebp",
   "version": "1.6.0",
+  "port-version": 1,
   "description": "WebP codec: library to encode and decode images in WebP format",
   "homepage": "https://github.com/webmproject/libwebp",
   "license": "BSD-3-Clause",
@@ -33,7 +34,10 @@
   "default-features": [
     "libwebpmux",
     "nearlossless",
-    "simd"
+    {
+      "name": "simd",
+      "platform": "!emscripten"
+    }
   ],
   "features": {
     "all": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5630,7 +5630,7 @@
     },
     "libwebp": {
       "baseline": "1.6.0",
-      "port-version": 0
+      "port-version": 1
     },
     "libwebsockets": {
       "baseline": "4.3.5",

--- a/versions/l-/libwebp.json
+++ b/versions/l-/libwebp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c0a4a598eacd380393ff3a92be045f4474643649",
+      "version": "1.6.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "d9dee7fa77c6779e018d20afd7f216813dc33f7c",
       "version": "1.6.0",
       "port-version": 0

